### PR TITLE
Fix `useAsyncOptionsPropsSelect` Select story

### PR DESCRIPTION
**Pull request by fichtnerma**
Fix a bug inside useAsyncOptionsPropsSelect that caused selected options to be overwritten by the initialValues on each render by memoizing the initialValues variable.

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-970
-   [x] Provide screenshots/screencasts if the change contains visual changes

### CODE REVIEW

File: storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx [hunk 1]
```diff
@@ -23,14 +23,16 @@ storiesOf("stories/hooks/useAsyncOptionsPropsSelect", module).add("Select", () =
...
+            <Form onSubmit={() => {}} keepDirtyOnReinitialize initialValues={{ selectAsync: { value: "strawberry", label: "Strawberry" } }}>
                 {() => (
                     <Field
                         name="selectAsync"
                         component={FinalFormSelect}
                         {...selectAsyncProps}
                         getOptionLabel={(option: Option) => option.label}
                         getOptionSelected={(option: Option, value: Option) => {
+                            console.log({ option, value });
+
```
Comment (johnnyomair) [inline storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx:35]: Come on 😁

**Review by johnnyomair — CHANGES_REQUESTED** [review]
Please provide more information in the PR's description. It should be possible to understand what you're doing without having to look at the JIRA task. Also, describe what and how you're fixing. This could then be the commit message of the squashed commit, making the change more comprehensible.

File: storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx [hunk 2]
```diff
@@ -23,14 +23,16 @@ storiesOf("stories/hooks/useAsyncOptionsPropsSelect", module).add("Select", () =
 
     return (
         <div style={{ minHeight: "50px" }}>
-            <Form onSubmit={() => {}} initialValues={{ selectAsync: { value: "strawberry", label: "Strawberry" } }}>
+            <Form onSubmit={() => {}} keepDirtyOnReinitialize initialValues={{ selectAsync: { value: "strawberry", label: "Strawberry" } }}>
```
Comment (johnnyomair) [inline storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx:26]: Please try to fix this differently, either bei memo'ing the initialValues, or declaring the options objects in a variable and using the same object for the initial value. See https://github.com/vivid-planet/comet/blob/main/storybook/src/docs/form/components/stories/FinalFormFields.stories.tsx#L69-L84

File: storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx (hunk 1 above)
Comment (fichtnerma) [reply storybook/src/docs/hooks/useAsyncOptionsProps/useAsyncOptionsPropsSelect.stories.tsx]: Sorry must have missed that

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**